File: apps/api/src/main/java/com/yalabs/yahub/shared/Result.java
package com.yalabs.yahub.shared;

public record Result<T>(
    boolean result,
    String message,
    T data
) {
    public static <T> Result<T> success(T data) {
        return new Result<>(true, null, data);
    }

    public static <T> Result<T> failure(String message) {
        return new Result<>(false, message, null);
    }
}

File: Back-end/src/YaHub.Application/Common/Result.cs
using System.Text.Json.Serialization;

namespace YaHub.Application.Common;

public sealed record Result<T>(
    [property: JsonPropertyName("result")] bool IsSuccess,
    string? Message,
    T? Data)
{
    public static Result<T> Ok(T data)
    {
        return new Result<T>(true, null, data);
    }

    public static Result<T> Fail(string message)
    {
        return new Result<T>(false, message, default);
    }
}
